### PR TITLE
files structure: move src_paths as a dynamic property, add remove

### PR DESCRIPTION
**Pull request by ultrabug**
Plugins wishing to manipulate the Files structure have an append
method but can't dynamically work on src_paths since it's computed
once on Files instanciation

This commits allows plugins to manipulate Files trough append and
remove of file objects while not having to care about the src_paths
since it's now a dynamically computed property

### CODE REVIEW

File: mkdocs/structure/files.py [hunk 1]
```diff
@@ -35,6 +38,10 @@ def append(self, file):
         self._files.append(file)
         self.src_paths[file.src_path] = file
```
Comment (waylan) [inline mkdocs/structure/files.py:39]: This line will need to be removed as it would not generate an error.

**Review by waylan — CHANGES_REQUESTED** [review]
This looks good. However, we should have some tests added. That fact that we are not getting any failing tests with the below noted issue suggest that we need to add some.

**Comment by ultrabug** [comment]
Thanks @waylan ; tests added

**Comment by ultrabug** [comment]
Added mention to release notes; hopefully to the right place

Cheers

**Review by waylan — APPROVED** [review]

**Pull request merged**